overlord/snapstate/snapstate.go

```diff
@@ -200,7 +200,32 @@ func doInstall(s *state.State, snapst *SnapState, ss *SnapSetup) (*state.TaskSet
 		addTask(s.NewTask("cleanup", fmt.Sprintf("Clean up %q%s install", ss.Name(), revisionStr)))
 	}
 
-	return state.NewTaskSet(tasks...), nil
+	var defaults map[string]interface{}
+
+	if ss.SideInfo != nil && ss.SideInfo.SnapID != "" {
+		gadget, err := GadgetInfo(s)
+		if err != nil && err != state.ErrNoState {
+			return nil, err
+		}
+		if err == nil {
+			gadgetInfo, err := snap.ReadGadgetInfo(gadget)
+			if err != nil {
+				return nil, err
+			}
+			defaults = gadgetInfo.Defaults[ss.SideInfo.SnapID]
```

Comment:
Indeed.. I'll write some unit tests for this, potentially in a follow up, and I'd also like to have proper spread tests with a real gadget we can exercise this logic with.